### DISCUSSION TITLE
What does Required mean?

### DISCUSSION THREAD

**Discussion by kbabiy** [tom-englert/Nullable.Extended#6] [Q&A] [answered]
[Updated] - started this as a question, then guessed and will leave a suggestion

Hi @tom-englert 

Thank you for the amazing extension you have created

For the "Required" binary field (with tick boxes) in the UI - a very useful field, though I'd suggest to highlight its meaning in the document or as a tool tip in the UI (made me wonder for a while)

Thank you!

**Comment by tom-englert** [comment, marked as answer]
Good idea, maybe you can contribute that?

**Reply by kbabiy** [reply]
I would, unless my VS2019 got bricked. The only thing I recently changed is installing this extension, so it's the main suspect :)

**Reply by tom-englert** [reply]
I'm using it frequently in VS2019 and VS2022, never had such a problem. It really does nothing unless you actively open it's window.

**Reply by tom-englert** [reply]
P.S. you can edit the readme directly in the browser...

**Reply by kbabiy** [reply]
yes, so I've docked the window and it was loading on every start of VS2019

Now after clean reinstall of VS and adding the extension, VS loads as the window is not docked.

I am trying to open the window and it crashes VS. Could you please check the logs?

[log.txt](https://github.com/tom-englert/Nullable.Extended/files/6983212/log.txt)

It looks like the extension is trying try load the component it can't find.

Maybe you could point me where should I look?

Thank you

**Reply by kbabiy** [reply]
This is what I managed to get with Sysinternals ProcMon
```

17:39:21.2793192	devenv.exe	2112	QueryDirectory	C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Ninject.Extensions.*.dll	NO SUCH FILE	FileInformationClass: FileBothDirectoryInformation, Filter: Ninject.Extensions.*.dll
17:39:21.2798498	devenv.exe	2112	QueryDirectory	C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Ninject.Web*.dll	NO SUCH FILE	FileInformationClass: FileBothDirectoryInformation, Filter: Ninject.Web*.dll

```

Some mistery why would it look for a component in IDE folder....

**Reply by kbabiy** [reply]
Tried VS2022 - clean install on my system (didn't have it), then install the extension and it crashes VS (same error logs)

I give up, doesn't seem I can help this problem or I have a chance to use the tool as it is now.

**Reply by tom-englert** [reply]
Created issue #7. 
Seems this was hidden for me due to some side effects, did not occur when other extensions were already installed.

**Reply by tom-englert** [reply]
@kbabiy can you try https://www.vsixgallery.com/extensions/Nullable.Extended.75a92614-c590-4401-b04b-04926c0e21cf/Nullable.Extended%20v1.5.4491.vsix to verify that this fixes the issue?

**Reply by kbabiy** [reply]
Hi @tom-englert 

Yes, it did. Thank you!

**Reply by tom-englert** [reply]
Thanks for helping to investigate this :+1: :smile:

**Reply by kbabiy** [reply]
Thank you for your help fixing the problem. 

I've also proposed the change to the README.md regarding the original purpose of the topic

**Reply by tom-englert** [reply]
Already did some small improvement: https://github.com/tom-englert/Nullable.Extended/commit/a909516fae97f3905332031044db74d171d7d3b7
